### PR TITLE
Add back support for reading from parent command, config file, or subcommand in flyte-cli

### PR DESCRIPTION
**Pull request by wild-endeavor**
# TL;DR
If not set on the command line, flyte-cli was not reading the host/insecure flags from the config file.  Behavior on master is:

```bash
(flytekit) ytong@argus:~/go/src/github.com/flyteorg/flytekit [flyte-sandbox] (master) $ cat ~/.flyte/local_sandbox
[platform]
url=localhost:30081
insecure=True

(flytekit) ytong@argus:~/go/src/github.com/flyteorg/flytekit [flyte-sandbox] (master) $ flyte-cli -c ~/.flyte/local_sandbox get-workflow -u wf:flytesnacks:development:core.control_flow.subworkflows.parent_wf:v0.3.56
DeprecationWarning: The command 'flyte-cli' is deprecated.
Config loading

################################################################################################################################
# flyte-cli is being deprecated in favor of flytectl. More details about flytectl in https://docs.flyte.org/projects/flytectl/ #
################################################################################################################################

Welcome to Flyte CLI! Version: 0.0.0+develop

Traceback (most recent call last):
  File "/Users/ytong/envs/flytekit/bin/flyte-cli", line 33, in <module>
    sys.exit(load_entry_point('flytekit', 'console_scripts', 'flyte-cli')())
  File "/Users/ytong/envs/flytekit/lib/python3.8/site-packages/click/core.py", line 1128, in __call__
    return self.main(*args, **kwargs)
  File "/Users/ytong/envs/flytekit/lib/python3.8/site-packages/click/core.py", line 1053, in main
    rv = self.invoke(ctx)
  File "/Users/ytong/envs/flytekit/lib/python3.8/site-packages/click/core.py", line 1659, in invoke
    return _process_result(sub_ctx.command.invoke(sub_ctx))
  File "/Users/ytong/envs/flytekit/lib/python3.8/site-packages/click/core.py", line 1395, in invoke
    return ctx.invoke(self.callback, **ctx.params)
  File "/Users/ytong/envs/flytekit/lib/python3.8/site-packages/click/core.py", line 754, in invoke
    return __callback(*args, **kwargs)
  File "/Users/ytong/go/src/github.com/flyteorg/flytekit/flytekit/clis/flyte_cli/main.py", line 809, in get_workflow
    client = _get_client(host, insecure)
  File "/Users/ytong/go/src/github.com/flyteorg/flytekit/flytekit/clis/flyte_cli/main.py", line 277, in _get_client
    return _friendly_client.SynchronousFlyteClient(cfg, root_certificates=parent_ctx.obj["cacert"])
  File "/Users/ytong/go/src/github.com/flyteorg/flytekit/flytekit/clients/raw.py", line 116, in __init__
    self._channel = grpc.secure_channel(
  File "/Users/ytong/envs/flytekit/lib/python3.8/site-packages/grpc/__init__.py", line 2004, in secure_channel
    return _channel.Channel(target, () if options is None else options,
  File "/Users/ytong/envs/flytekit/lib/python3.8/site-packages/grpc/_channel.py", line 1479, in __init__
    _common.encode(target), _augment_options(core_options, compression),
  File "/Users/ytong/envs/flytekit/lib/python3.8/site-packages/grpc/_common.py", line 72, in encode
    return s.encode('utf8')
AttributeError: 'NoneType' object has no attribute 'encode'

```

## Type
 - [x] Bug Fix
 - [ ] Feature
 - [ ] Plugin

## Are all requirements met?

 - [x] Code completed
 - [x] Smoke tested
 - [ ] Unit tests added
 - [x] Code documentation added
 - [ ] Any pending items have an associated Issue

## Complete description
* Add back creation of a separate context (separate from the `setup-config` command).
* If values for host/insecure are in the config object but not in the parent or sub command, then fill them in.
* Missing insecure in `with_parameters` in `PlatformConfig`.
* Don't pass root certs if creating an insecure client.

See the `make_context` code before the change [here](https://github.com/flyteorg/flytekit/blob/457f323c0052df822c962186f44a6989a418d23d/flytekit/clis/flyte_cli/main.py#L474).

### CODE REVIEW

**Review by kumare3 — APPROVED** [review]

**Comment by wild-endeavor** [comment]
sorry one more time?  had a lint

**Review by kumare3 — APPROVED** [review]
